SDLC Checklists (#7448)
auth0/docs: articles/architecture-scenarios/checklists.md

---
title: Implementation Checklists
description: Links to checklists for your implementation. 
topics:
    - SDLC
    - checklists
    - best practices
    - implementation checklist
contentType: reference
useCase:
    - implementation
---
# Implementation Checklists

> Auth0 recommends that you break your implementation into the following 6 phases to plan and complete your implementation. 

Reviewer: ```suggestion
Auth0 provides a number of Checklist (in Google Sheets format) that will help you co-ordinate progress as you work through each phase in your Software Development Life Cycle (SDLC).

For each SDLC phase, the relevant Checklist (see below) will provided you with a list of items you will need to consider, together with links to pertinent Auth0 feature documentation covering each. Not all items will necessary apply to the project you are working on, so we advise you to copy and modify a checklist to suit your needs.  
```

* [Analyze](#analyze)
* [Design](#design)
* [Build](#build)
* [Test](#test)
* [Deploy](#deploy) 
* [Monitor](#monitor)

> Auth0 provides a checklist (Google Sheets format) for each phase with links to pertinent Auth0 feature documentation. Not all parts of each phase may apply to you. You can copy and modify the checklists to suit your needs. 

Reviewer: ```suggestion

```

## Analyze

> Use the [Analyze checklist](https://docs.google.com/spreadsheets/d/1hTtRvvO6Szlvcr4XnsgzzmJj7FmmucAZAmFF275hB20/edit?usp=sharing) to analyze end-user business requirements and determine project goals as part of the high-level plan for the project. Convert the requirements and goals into system functions that the organization intends to develop. Activities include:

Reviewer: ```suggestion
Use the [Analyze checklist](https://docs.google.com/spreadsheets/d/1hTtRvvO6Szlvcr4XnsgzzmJj7FmmucAZAmFF275hB20/edit?usp=sharing) during the Analysis phase of your SDLC, where you'll analyze end-user business requirements and determine project goals as part of the high-level plan for the project. The Checklist for this phase will help you focus on Auth0 specifics as you convert the requirements and goals into system functions that you intend to develop.
```

> * Gathering business requirements

Reviewer: ```suggestion

```

> * Creating process diagrams

Reviewer: ```suggestion

```

> * Performing detailed analysis

Reviewer: ```suggestion

```

> * Alignment to project plan

Reviewer: ```suggestion

```

## Design

> Use the [Design checklist](https://docs.google.com/spreadsheets/d/1L0PLqcghVvCSdW_OmLoKlU0uDJtb5mw6uIufMrKW7ro/edit?usp=sharing) to describe the desired features and operations of the system, including  business rules, pseudo-code, screen layouts, and other necessary documentation. Activities include:

Reviewer: ```suggestion
Use the [Design checklist](https://docs.google.com/spreadsheets/d/1L0PLqcghVvCSdW_OmLoKlU0uDJtb5mw6uIufMrKW7ro/edit?usp=sharing) during the Design phase of the SDLC, to help you focus on Auth0 specifics as you describe the desired features and operations of the system, including business rules, pseudo-code, screen layouts, and any other necessary collateral.
```

> * Infrastructure design

Reviewer: ```suggestion

```

> * System model design

Reviewer: ```suggestion

```

## Build

> Use the [Build checklist](https://docs.google.com/spreadsheets/d/1Wd5HwONEi2Max7zcORJpwn8yQO_mbu62FDc-Sk8oWS8/edit?usp=sharing) to develop the actual system through implementation of infrastructure and code. Activities include:

Reviewer: ```suggestion
Use the [Build checklist](https://docs.google.com/spreadsheets/d/1Wd5HwONEi2Max7zcORJpwn8yQO_mbu62FDc-Sk8oWS8/edit?usp=sharing) during the Build phase of the SDLC, to help you focus on Auth0 specifics as you develop the actual system through implementation of infrastructure and code.
```

> * Infrastructure implementation

Reviewer: ```suggestion

```

> * Code implementation

Reviewer: ```suggestion

```

## Test

> Use the [Test checklist](https://docs.google.com/spreadsheets/d/19tQOQ_heBVpHXx-YKZZY20u6y8mU8IwRHEF32znZ_-w/edit?usp=sharing) to integrate and deploy all implemented code in the testing environment infrastructure. Testing then follows Software Testing Life Cycle activities to check the system for errors, bugs, and defects to verify that system functionalities work as expected (or not). Activities include:

Reviewer: ```suggestion
Use the [Test checklist](https://docs.google.com/spreadsheets/d/19tQOQ_heBVpHXx-YKZZY20u6y8mU8IwRHEF32znZ_-w/edit?usp=sharing) during the Test phase of the SDLC, to help focus on Auth0 specifics as you integrate and deploy all implemented code in your testing environment infrastructure and perform testing in line with your Software Testing Life Cycle activities (in order to check the system for errors, bugs, and defects).
```

> * Write test cases

Reviewer: ```suggestion

```

> * Execute test cases

Reviewer: ```suggestion

```

## Deploy

> Use the [Deploy checklist](https://docs.google.com/spreadsheets/d/1MRBz4CiT6RTL7Mf7eWmpjSZEQS3rwjNryyPGmvjn-fc/edit?usp=sharing) to deploy the system to either a staging or production environment, where actual users begin to operate and interact with it. 

Reviewer: ```suggestion
Use the [Deploy checklist](https://docs.google.com/spreadsheets/d/1MRBz4CiT6RTL7Mf7eWmpjSZEQS3rwjNryyPGmvjn-fc/edit?usp=sharing) during the Deployment phase of the SDCL, to help focus on Auth0 specifics as you prepare to deploy the system to either a staging or production environment, where actual users will begin to operate and interact with it. 
```

Eventually, you deploy all components of the system to the production environment as a live release is made.

## Monitor

> Use the [Monitor checklist](https://docs.google.com/spreadsheets/d/1zHs29OoeiludHxY5cV06oGVAIW91jMjUflJgzJVaRKo/edit?usp=sharing) to make enhancements, corrections, and changes to ensure the system continues to work and stays updated to meet the business objectives and support the needs of the users. Activities include: 

Reviewer: ```suggestion
Use the [Monitor checklist](https://docs.google.com/spreadsheets/d/1zHs29OoeiludHxY5cV06oGVAIW91jMjUflJgzJVaRKo/edit?usp=sharing) during the Monitoring phase of the SDLC, to help focus on Auth0 specifics as you make enhancements, corrections, and changes to your live production environment, in order to ensure the system continues to work as desired and stays updated to meet the business objectives as well as support the needs of the your users. 
```

> * Monitoring

Reviewer: ```suggestion

```

> * Maintenance

Reviewer: ```suggestion

```

> * Changes and adjustments

Reviewer: ```suggestion

```

> * Upgrade and adapt to future needs 

Reviewer: ```suggestion

```
